include/binmap/graph.hpp

```diff
@@ -72,7 +74,7 @@ class Graph {
   bool has_path(boost::filesystem::path const &from,
                 boost::filesystem::path const &to) const;
 
-  void add_node(boost::filesystem::path const &input_file,
+  boost::filesystem::path add_node(boost::filesystem::path const &input_file,
```

Comment:
Please document the behavior with respect to add_node return value here

Reply:
graph.add_node return a path which is used in scan.add_node (this one return the same path.)

It's then used in operator() : 
if the return of add_node was the same as input_file (that means it hasn't been parsed before and it is in the same path as the binary who had the dependency or it isn't in any path found before) we continue,

if the return of add_node gives an other binary : we look if it hasn't been parsed before. If it hasn't we continue and add to the visited_ (actually we add in both cases, because there isn't problem if we have the same bin twice in visited_ , but we can change it.)

After, we continue the analyze with the binary found. (analyze_dependencies, analyze_metadata, add_deps)

Reply:
ok, I meant to add this as a comment in the code, something like

```c++
/**
 * if `input_file` was not met before, add it to the graph and return it. Otherwise ...
 */
```